docs/content/devguide/097-rest.adoc

```diff
@@ -0,0 +1,63 @@
+
+=== Authentication
+The REST server provides a simple HTTP authentication mechanism to control access to the service. Upon starting for the first time, the server will create a passwords.csv file in the geowave config directory (i.e. `.geowave/passwords.csv`), with default credentials `admin` and `password`:
+
+[source, bash]
+----
+admin,5e884898da28047151d0e56f8dc6292773603d0d6aabbdd62a11ef721d1542d8 #SHA-256 of "password"
+----
+
+To add another user, modify this file and add the user and password hash of the new user (a future command may be implemented to do this more easily), and restart the server.
+
+=== Support for New Commands
+To enable access to the REST server for a new command, perform the following steps:
+
+. Ensure that the `geowave-service-rest` project has a dependency on the new command. All classes which the REST server can access are visible from the homepage at `http://localhost:5152`, and should appear in [blue]#blue# if they are not yet implemented. 
+. Ensure that the class implementing the operation extends `DefaultOperation`. This is necessary because `DefaultOperation` extends `ServerResource`, which restlet requires for all endpoints.
+. Ensure that the `@GeoWaveOperation` annotation has `restEnabled=RestEnabledType.GET` or `restEnabled=RestEnabledType.POST`, depending on whether the operation is a mutating or non-mutating change (see http://blog.teamtreehouse.com/the-definitive-guide-to-get-vs-post[this article] for more info).
+. In order for parameters to be correctly processed, the http://jcommander.org/#_main_parameter[main parameter], if present, must be annotated with a `@RestParameters` annotation. These names correspond to the fields on the query string (for GET requests) or in the form (for POST requests). All other operations use the same name as the `@Parameter` field. For example,
+
+[source, java]
+----
+// CopyStoreCommand
+
+@Parameter(description = "<name> <new name>")
+@RestParameters(names = {
+	"name",
+	"newname"
+})
+private List<String> parameters = new ArrayList<String>();
+
+@Parameter(names = {
+	"-d",
+	"--default"
+}, description = "Make this the default store in all operations")
+private Boolean makeDefault;
+----
+
+Sample request:
+
+[source]
+----
+name=storetest
+newname=storetesting
+makeDefault=true
+----
+
+[start=5]
+. Common code that is invoked from the http request should be implemented in the operation's `computeResults` method (inheriting from `DefaultOperation#computeResults`). This is the method that will be called with the operation is invoked via REST, and should return a Bean object that will be serialized to JSON for the response. If the operation produces no response (for example, in the case of `CopyStoreCommand`), it should return `null`, which can be enforced by specifying a return type of https://docs.oracle.com/javase/8/docs/api/java/lang/Void.html[Void]. The execute method should call `computeResults`, and then display the results (if any) to the console.
```

Comment:
might be worth mentioning that the computeResults logic can be taken directly from the existing execute and then just having execute call that. this is hinted at with 'the execute method shoudl call computeResults' but it might be nice to be more explicit about this, as it could save people a lot of time from reimplementing execute